### PR TITLE
fix(linter): rules-of-hooks fix false positive with default export

### PR DESCRIPTION
**Pull request by camc314**
closes #7555

### CODE REVIEW

**Comment by camc314** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/7570?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#7570** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/7570?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈 <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/7570?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* **#7569** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/7569?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`

<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.

File: crates/oxc_linter/src/rules/react/rules_of_hooks.rs [hunk 1]
```diff
@@ -172,13 +172,12 @@ impl Rule for RulesOfHooks {
             }) => {
                 let ident = get_declaration_identifier(nodes, parent_func.id());
 
-                // Hooks cannot be called inside of export default functions or used in a function
-                // declaration outside of a react component or hook.
+                // Hooks cannot be used in a function declaration outside of a react component or hook.
```
Comment (camc314) [inline crates/oxc_linter/src/rules/react/rules_of_hooks.rs:175]: cc @rzvxa looks like you added this comment originally. but, as far as i understand there is nothing incorrect about calling a hook within a function that's default exported + react doesn't report

let me know if there was a reason this should exist - thanks!

**Comment by camc314** [comment]
@Boshen i think we shuold promote this one to correctness after running in ecosystem CI

**Comment by Boshen** [comment]
### Merge activity

* **Dec 1, 10:08 PM EST**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.dev/merges?org=oxc-project&repo=oxc) once it meets the requirements.

**Pull request merged**